src/state/opamEnv.ml

```diff
@@ -639,14 +639,21 @@ let update_user_setup root ?dot_profile shell =
     OpamStd.Option.iter (fun f -> update_dot_profile root f shell) dot_profile
   )
 
+let status st =
+  OpamStateConfig.update ~no_env_notice:false ();
+  not
+    (OpamFile.Config.switch st.switch_global.config = Some st.switch ||
+     OpamStateConfig.(!r.switch_from <> `Command_line)) ||
```

Comment:
Why these conditions ? This would lead e.g. `opam env --check --switch foo` to always return 0, which is not what I would expect. Just `is_up_to_date` seems to be what you want here, maybe aside from the `no_env_notice` setting...